packages/callout-block/src/CalloutBlock.tsx

```diff
@@ -35,23 +37,19 @@ export const CalloutBlock = ({ appBridge }: BlockProps): ReactElement => {
         blockSettings.width === Width.HugContents && alignmentMap[blockSettings.alignment],
     ]);
 
-    const getAccentColor = (type: Type): string => {
-        const style = getComputedStyle(document.body);
-        switch (type) {
-            case Type.Info:
-                return style.getPropertyValue(`${THEME_PREFIX}accent-color-info-color`);
-            case Type.Note:
-                return style.getPropertyValue(`${THEME_PREFIX}accent-color-note-color`);
-            case Type.Tip:
-                return style.getPropertyValue(`${THEME_PREFIX}accent-color-tip-color`);
-            case Type.Warning:
-                return style.getPropertyValue(`${THEME_PREFIX}accent-color-warning-color`);
-        }
-    };
+    useEffect(() => {
+        const updateStyles = () => {
+            const { backgroundColor, textColor } = computeStyles(type, appearance);
+            setBackgroundColor(backgroundColor);
+            setTextColor(textColor);
+        };
+
+        window.emitter.on('HubAppearanceUpdated', () => {
```

Comment:
Will this create a new subscription every time this useEffect runs? I suggest to unsubscribe in the return function of this useEffect.

Reply:
You are completely right, missed that one